python/hidet/ir/expr.py

```diff
@@ -561,7 +562,7 @@ def __init__(self, hint: Optional[str], type: BaseType, name: Optional[str] = No
         self.hint: Optional[str] = hint
         self.name: Optional[str] = name
         self.type: Union[BaseType, TensorType, TensorPointerType, FuncType] = type
-        self.id: int = self.new_id()
+        self.id: int = self.new_id() if hidet.option.get_option('debug_show_var_id') else 0
```

Comment:
It is better to put the logic into the `new_id` function to make the `__init__` looks more clean.